### ISSUE TITLE
SCORM Runtime

### ISSUE THREAD

**Issue by arashster** [Mustufask/Scorm#1]
Hi, just saw your previous message on scorm-again. Are you trying to do SCORM tracking?

**Comment by Mustufask** [comment]
@arashster  yes do u have any approach?

**Comment by arashster** [comment]
Yes you can use scorm-again. But what are you trying to track? Just completion status (like “passed”, “completed”, “failed”, “incomplete”)? Score? All of that would be in the CMI data.

Are you integrating this into your own LMS system?

**Comment by arashster** [comment]
Are you looking for something like this?

**Comment by Mustufask** [comment]
sir yes can u pls share me the logic

**Comment by arashster** [comment]
Email me at arashster@gmail.com

**Comment by arashster** [comment]
To implement this, you need to do a bunch of stuff:

- unzip your SCORM package and make it accessible on your web server
- Set up a page that includes the scorm-again library as well as use the right runtime library (e.g. player.html), and initialize a CMI Object and the SCORM Runtime API (e.g. window.API = new Scorm12API(settings). Your package appears to be a SCORM 1.2 package so this should be sufficient.
- Parse the imsmanifest.xml file and find all the activities that the package has, and create a frame or a link to the popup window that opens that activity page. You can do this on the player.html page. That package will AUTOMATICALLy look for the window.API object on the player.html page.
- Add the event listeners for the right objects, eg. window.API.on("LMSSetValue.cmi.*", function(CMIElement, value) {
  [...]
});

You need to understand the SCORM CMI, a good reference is here: https://scorm.com/scorm-explained/technical-scorm/run-time/run-time-reference/

**Comment by Mustufask** [comment]
@arashster can u pls share me your above implementation in my scorm package through mail: mustufask54@gmail.com ?
bcz im a beginner with scorm, i would find the server url in it.
can u pls share

**Comment by Mustufask** [comment]
@arashster,
Thanks for yr approach, scorm again works well with my scorm1.2 of only quiz package but sometime for my another quiz package all interactions doesnt captures only cmi_suspend.data works and after completion if status is failed then also the status shows passed and score also doesnt updates here's my code :
<html>
<head>
	<title>SCORM 1.2</title>
	<script src="https://code.jquery.com/jquery-3.5.1.min.js"></script>
	<script src="./node_modules/scorm-again/dist/scorm-again.js"></script>

	<script>
		let settings = {
			"autocommit": true,
			alwaysSendTotalTime: true,
			asyncCommit: true,
			autoProgress: true,
			commitRequestDataType: "application/json;charset=UTF-8",
			dataCommitFormat: "json",
			lmsCommitUrl: "http://localhost:3001/xapi/statements",
			logLevel: 4,
			mastery_override: false
		}

		window.API = new Scorm12API(settings);

		window.API.on("LMSSetValue.cmi.*", function (CMIElement, value) {
			//Current CMI Object
			console.log('CMI Element', CMIElement)
			console.log('value', value)
			console.log('SCORE:', window.API.cmi.core.score.raw);
			console.log('STATUS:', window.API.cmi.core.lesson_status);
		});
	</script>
</head>

<body>
	<iframe src="Spot the fish/index_lms.html" style="width: 100vw; height: 100vh;"></iframe>
</body>

</html>

Also, if i use my scorm package with only slides then also my lesson_location and lesson_status returns null so how can i fix it

**Comment by Mustufask** [comment]
[Spot the fish.zip](https://github.com/user-attachments/files/16575603/Spot.the.fish.zip)
my scorm package attached
